pkg/odo/cli/component/common_link.go

```diff
@@ -172,6 +179,35 @@ func (o *commonLinkOptions) validate(wait bool) (err error) {
 			return fmt.Errorf("Couldn't find service named %q. Refer %q to see list of running services", svcFullName, "odo service list")
 		}
 
+		if o.operationName == unlink {
+			componentName := o.EnvSpecificInfo.GetName()
+			sbrName := getSBRName(componentName, o.serviceType, o.serviceName)
+			links := o.EnvSpecificInfo.GetLink()
+
+			linked := isComponentLinked(sbrName, links)
+			if !linked {
+				// user's trying to unlink a service that's not linked with the component
+				return fmt.Errorf("failed to unlink the service %q since it's not linked with the component %q", svcFullName, componentName)
+			}
+
+			//verify if the underlying service binding request actually exists
+			sbrSvcFullName := strings.Join([]string{sbrKind, sbrName}, "/")
+			sbrExists, err := svc.OperatorSvcExists(o.KClient, sbrSvcFullName)
+			if err != nil {
+				return err
+			}
+			if !sbrExists {
+				// this could have happened if the service binding request was deleted outside odo workflow (eg: oc delete sbr/<sbr-name>)
+				// we must remove entry of the link from env.yaml in this case
+				err = o.Context.EnvSpecificInfo.DeleteLink(sbrName)
+				if err != nil {
+					return fmt.Errorf("component's link with %q has been deleted outside odo; unable to delete odo's state of the link", svcFullName)
+				}
+				return fmt.Errorf("component's link with %q has been deleted outside odo", svcFullName)
```

Comment:
`components`

Reply:
I think `component's` is just right here since we're talking about a specific component. :coffee: :wink: